### DISCUSSION TITLE
AI order 'Load Items (Load Ammo)' inconsistency w/ modded weapons. Tag or bug?

### DISCUSSION THREAD

**Discussion by Ek-Videogames** [FakeFishGames/Barotrauma#9519] [Code questions and modding support]
I've run into a rather annoying issue trying to get AI 'Reload Ammo' order to work with the custom hardpoints of EK.
And this is the strangest thing:
- **The AI will reload them correctly when given the "operate weapons" command**. Only the new "reload items" command isn't working.
- All loaders that use the default "turretammosource" tag seem to work, but custom hardpoints with a unique shared-tag don't.
`( however checking the source code it seems AI derives item targets from the "Containable" part of the XML - turretammosource is not used anywhere - so if this is the cause of the issue it still doesn't seem to make sense? )`
If the tag is somehow integral to the order AI or some other function that still seems to be an oversight, as the hardpoint system included changes in modding specifically to facilitate making 'unique' hardpoint category tag sets.

So I'm not sure if there's an XML issue somewhere I'm missing or if this is warranting a modding issue ticket?
If the latter, you can move it to a ticket - create a new ticket - or simply let me know and I will compile the information here into the ticket format so it's less work for you guys.

### Example excerpt from mod XML: [AI loader tag issue.xml](https://github.com/Regalis11/Barotrauma/files/9035402/AI.loader.tag.issue.xml.txt)

![image](https://user-images.githubusercontent.com/55356135/177055087-f84745b7-4f30-47c5-a6a6-02df7ce91d02.png)
### Steps To Reproduce:
Test example submarine file: [EkGunnery AI Loader Debug.zip](https://github.com/Regalis11/Barotrauma/files/9035409/EkGunnery.AI.Loader.Debug.zip)
- (EK Gunnery Required): [Steam Workshop Page](https://steamcommunity.com/sharedfiles/filedetails/?id=2390126010)
- Order AI crewmembers to "Load Items (Load Ammo)"
Crew AI will load all the guns on the left side but not the right.
- Order AI crewmembers to "Operate Weapons"
Crew AI will load a single-item of ammo correctly on both the left & right weapons.

**Comment by T1MM5H** [comment]
I think you should make this an Issue instead of a Discussion, since this seems to be a code problem with the way the AI checks for Ammo Status.

**Comment by Ek-Videogames** [comment]
The quirk with bots only loading one shell into loaders even when they have multiple slots seems to be the same as this previously mentioned issue:

_https://github.com/Regalis11/Barotrauma/issues/8305#issuecomment-1047266051_
> It seems like the bots now are loading shells into the railgun loader, but only a single one. When that one is fired, at some point another single shell is loaded, never the full railgun loader to max capacity.
> 
> Unstable V0.16.4.0
